fs/bcachefs/fs-io.c

```diff
@@ -2307,16 +2307,17 @@ int bch2_truncate(struct user_namespace *mnt_userns,
 	int ret = 0;
 
 	/*
-	 * Don't update timestamps if we're not doing anything:
+	 * If the truncate call with change the size of the file, the
+	 * cmtimes should be updated. If the size will not change, we
+	 * do not need to update the cmtimes.
 	 */
-	if (iattr->ia_size == inode->v.i_size)
-		return 0;
-
-	if (!(iattr->ia_valid & ATTR_MTIME))
-		ktime_get_coarse_real_ts64(&iattr->ia_mtime);
-	if (!(iattr->ia_valid & ATTR_CTIME))
-		ktime_get_coarse_real_ts64(&iattr->ia_ctime);
-	iattr->ia_valid |= ATTR_MTIME|ATTR_CTIME;
+	if (iattr->ia_size != inode->v.i_size) {
+		if (!(iattr->ia_valid & ATTR_MTIME))
+			ktime_get_coarse_real_ts64(&iattr->ia_mtime);
+		if (!(iattr->ia_valid & ATTR_CTIME))
+			ktime_get_coarse_real_ts64(&iattr->ia_ctime);
+		iattr->ia_valid |= ATTR_MTIME|ATTR_CTIME;
+	}
```

Comment:
Have you verified this? I think we want to do the opposite here - if i_size isn't changing, clear mtime/ctime from ia_valid

Reply:
This block is for if the i_size is changing, but I think you're right that I might have to clear [cm]time from the ia_valid in the else case.